ui/js/services/tcactions.js

```diff
@@ -39,48 +40,46 @@ treeherder.factory('tcactions', [
+                        const match = job.reason.match(/Created by BBB for task (.{22})/);
+                        if (match) {
+                            originalTaskId = match[1];
+                        }
+                    }
+                    originalTaskPromise = $http.get('https://queue.taskcluster.net/v1/task/' + originalTaskId).then(response => response.data);
+                }
+
+                return $q.all([
+                    $http.get(actionsUrl),
+                    originalTaskPromise,
+                ]).then(([response, originalTask]) => {
                     if (!response.data) {
                         // This is a push with no actions.json so we should
                         // allow an implementer to fall back to actions.yaml
                         return null;
                     }
-
                     if (response.data.version !== 1) {
                         thNotify.send("Wrong version of actions.json, can't continue", "danger", true);
                         return;
                     }
 
-                    let originalTaskPromise = $q.resolve(null);
-                    let originalTaskId;
-                    if (job) {
-                        if (job.taskcluster_metadata) {
-                            originalTaskId = job.taskcluster_metadata.task_id;
-                        } else {
-                            // This is a bbb job in this case. We'll try our best.
-                            const match = job.reason.match(/Created by BBB for task (.{22})/);
-                            if (match) {
-                                originalTaskId = match[1];
-                            }
-                        }
-                        originalTaskPromise = $http.get('https://queue.taskcluster.net/v1/task/' + originalTaskId).then(
-                            function (response) {
-                                return response.data;
-                            });
-                    }
-                    return originalTaskPromise.then(originalTask => ({
+                    return {
                         originalTask,
                         originalTaskId,
                         staticActionVariables: response.data.variables,
-                        actions: response.data.actions.filter(function (action) {
-                            return action.kind === 'task' && (!originalTask || (
-                                !action.context.length || _.some((action.context).map(function (actionContext) {
-                                    return !Object.keys(actionContext).length || _.every(_.map(actionContext, function (v, k) {
-                                        return (originalTask.tags[k] === v);
-                                    }));
-                                }))));
-                        }),
-                    }));
+                        actions: response.data.actions.filter(action => action.kind === 'task' &&
```

Comment:
would you write up a comment for this section, too.  Just explaining this decision tree a bit.